Filter biobliography results by publication date

remo87 opened the pull request:
update total to include date filter
update minDate to use no limit query

remo87 committed:
update total query to include date filter
update minDate to use no limit query

remo87 committed:
update query to re use date filter

JarrodBaker reviewed (changes requested):
Just a small change about Option. Otherwise it's looking really good. I'd like us to try and get some unit testing around this. Let's discuss next Thursday how we can test this to a) have some tests as documentation and b) make sure we don't introduce any regressions in future when we work in this part of the code.

remo87 committed:
update date filter location due to error in filtering

JarrodBaker committed:
Update Literature queries to get total and year in one go.

There does not seem to be any advantage to use two queries to get the
total number of publications and the earliest publication date.

JarrodBaker committed:
Run scalafmt

JarrodBaker committed:
Remove unused code

JarrodBaker committed:
Simplify literature query logic

JarrodBaker committed:
Refactor to simplify common pre-where logic

JarrodBaker committed:
Add documentation

remo87 committed:
formart documents using scalafmt
add null folder to gitignore

remo87 committed:
merge master into branch

remo87 committed:
bring QLITAGG changes from master
reformat files
